meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -136,27 +139,63 @@ public void joinLeaderElectionParticipantPool(String leaderPath) {
   public void joinLeaderElectionParticipantPool(String leaderPath, LeaderInfo userInfo) {
     // TODO: create participant entry with info
     subscribeAndTryCreateLeaderEntry(leaderPath);
+
+    LeaderInfo participantInfo = new LeaderInfo(userInfo);
+    createParticipantInfo(leaderPath, participantInfo);
+  }
+
+  private void createParticipantInfo(String leaderPath, LeaderInfo participantInfo) {
+
+    try {
+      // try to create participant info entry, assuming leader election group node is already there
+      _metaClient.create(leaderPath + PARTICIPANTS_ENTRY_PARENT + _participant, participantInfo,
+          MetaClientInterface.EntryMode.EPHEMERAL);
+    } catch (MetaClientNodeExistsException ex) {
+      LOG.info("Already joined pool {}.", leaderPath);
```

Comment:
If this is reconnect, you dont need to recreate. It only happens when create a new session but old session was there.

Then why not throw the exception? When this happened, it usually means the old session is not gone. Throw exception let user handle it instead of swallowing it. Otherwise, if the old session is gone and you are not creating new session one, there could be potential problem.